### PR TITLE
create thumb image

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/265

Task #328: explore a project
 
Task #332: add isBigImage to imageRequestMapper 
 
Task #333: change ImageCommand
 
Task #334: update tests
 
Task #335: add pull request & fix bugs

### CODE REVIEW

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs [hunk 1]
```diff
@@ -108,15 +113,26 @@ public void SetUp()
...
-        public void ShouldAddNewImageAndThumbImmage()
+        public void ShouldAddThumbImmageWhenImageIsBig()
+        {
+            Assert.AreEqual(firstDbImage.Id, command.Execute(imageRequest));
+
+            validatorMock.Verify(v => v.Validate(It.IsAny<IValidationContext>()), Times.Once);
+            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Exactly(1));
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, out isBigImage, null), Times.Once);
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, out isBigImage, firstDbImage.Id), Times.Once);
+            //mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), It.IsAny<ImageType>(), out isBigImage, It.IsAny<Guid?>()), Times.Exactly(1));
```
Comment (BelehovEgor) [inline test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:124]: delete

File: src/FileService.Business/AddNewImageCommand.cs [hunk 2]
```diff
@@ -12,30 +12,35 @@ namespace LT.DigitalOffice.FileService.Business
...
+        private readonly IImageRepository _repository;
+        private readonly IValidator<ImageRequest> _validator;
+        private readonly IImageRequestMapper _mapper;
         public AddNewImageCommand(
-            [FromServices] IImageRepository repository,
-            [FromServices] IValidator<ImageRequest> validator,
-            [FromServices] IImageRequestMapper mapper)
+               IImageRepository repository,
+               IValidator<ImageRequest> validator,
+               IImageRequestMapper mapper)
```
Comment (KostinP) [inline src/FileService.Business/AddNewImageCommand.cs:21]: Extra spaces

**Review by KostinP — CHANGES_REQUESTED** [review]
We are waiting for the tests to be corrected

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs (hunk 1 above)
Comment (borovHokage) [reply test/FileService.Business.UnitTests/AddNewImageCommandTests.cs]: fixed

File: src/FileService.Mappers/RequestMappers/ImageRequestMapper.cs [hunk 3]
```diff
@@ -42,6 +45,7 @@ public DbImage Map(ImageRequest imageRequest, ImageType imageType, Guid? parentI
             }
             else
             {
+                isBigImage =false;
```
[the remark this answers is not in the document]
Comment (borovHokage) [reply src/FileService.Mappers/RequestMappers/ImageRequestMapper.cs:49]: fixed

File: src/FileService.Business/AddNewImageCommand.cs (hunk 2 above)
Comment (borovHokage) [reply src/FileService.Business/AddNewImageCommand.cs]: fixed

File: src/FileService.Mappers/RequestMappers/ImageRequestMapper.cs [hunk 4]
```diff
@@ -33,15 +34,19 @@ public DbImage Map(ImageRequest imageRequest, ImageType imageType, Guid? parentI
 
             if (imageType == ImageType.Full)
             {
+                using Image image = Image.Load(imageRequest.Content);
+                isBigImage = image.Width > 150 && image.Height > 150;
```
Comment (TIove) [inline src/FileService.Mappers/RequestMappers/ImageRequestMapper.cs:38]: What if image will have long size only on Height or Weight? like Height = 1000 and Weight =130
Is it ok image or not?

Comment (borovHokage) [reply src/FileService.Mappers/RequestMappers/ImageRequestMapper.cs:38]: no, we need only when both parametrs fulfil the conditions

**Review by MariyaSin — APPROVED** [review]

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs [hunk 5]
```diff
@@ -108,15 +114,36 @@ public void SetUp()
...
+        public void ShouldNotAddNewImageWhenImageIsNotBig()
         {
             Assert.AreEqual(firstDbImage.Id, command.Execute(imageRequest));
 
             validatorMock.Verify(v => v.Validate(It.IsAny<IValidationContext>()), Times.Once);
-            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Exactly(2));
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, null), Times.Once);
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, firstDbImage.Id), Times.Once);
-            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), It.IsAny<ImageType>(), It.IsAny<Guid?>()), Times.Exactly(2));
+            repositoryMock.Verify(r => r.AddNewImage(It.IsAny<DbImage>()), Times.Exactly(1));
```
Comment (KostinP) [inline test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:144]: Times.Once

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs [hunk 6]
```diff
@@ -108,15 +114,36 @@ public void SetUp()
...
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Full, out isBigImage, It.IsAny<Guid?>()), Times.Once);
+            mapperMock.Verify(m => m.Map(It.IsAny<ImageRequest>(), ImageType.Thumb, out isBigImage, It.IsAny<Guid?>()), Times.Never);
         }
 
         [Test]
```
Comment (KostinP) [inline test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:149]: _validatorMock
                .Setup(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
                .Returns(false)

**Review by KostinP — CHANGES_REQUESTED** [review]

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs (hunk 6 above)
Comment (borovHokage) [reply test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:148]: fixed

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs (hunk 5 above)
Comment (borovHokage) [reply test/FileService.Business.UnitTests/AddNewImageCommandTests.cs]: fixed

**Review by KostinP — APPROVED** [review]

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs [hunk 7]
```diff
@@ -108,48 +113,55 @@ public void SetUp()
...
-        public void ShouldAddNewImageAndThumbImmage()
+        public void ShouldAddThumbImageWhenImageIsBig()
         {
+            isBigImage = true;
```
Comment (djino97) [inline test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:118]: remove

**Review by djino97 — CHANGES_REQUESTED** [review]
Where edit tests of ImageRequestMapper?

**Review by TIove — APPROVED** [review]

File: src/FileService.Business/AddNewImageCommand.cs [hunk 8]
```diff
@@ -12,30 +12,35 @@ namespace LT.DigitalOffice.FileService.Business
...
+            var parentDbImage = _mapper.Map(request, ImageType.Full, out bool isBigImage);
 
+            if (isBigImage)
+            {
+                var childDbImage = _mapper.Map(request, ImageType.Thumb, out isBigImage, parentDbImage.Id);
+                _repository.AddNewImage(childDbImage);
+            }
+            else
+            {
+                isBigImage = false;
```
[the remark this answers is not in the document]
Comment (borovHokage) [reply src/FileService.Business/AddNewImageCommand.cs:41]: without  it, this condition doesn't work

File: test/FileService.Business.UnitTests/AddNewImageCommandTests.cs (hunk 7 above)
Comment (borovHokage) [reply test/FileService.Business.UnitTests/AddNewImageCommandTests.cs:118]: this is local condition for this test. without it test doesn't work, we did this on scrum on monday

File: src/FileService.Business/AddNewImageCommand.cs [hunk 9]
```diff
@@ -12,30 +12,35 @@ namespace LT.DigitalOffice.FileService.Business
...
+                _repository.AddNewImage(parentDbImage);
+            }
```
[the remark this answers is not in the document]
Comment (borovHokage) [reply src/FileService.Business/AddNewImageCommand.cs:44]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by djino97 — APPROVED** [review]

**Pull request merged**